src/menu/common.c

```diff
@@ -91,6 +91,7 @@ static void reset_game(StartGameContext *ctx) {
 	replay_destroy(&global.replay);
 	replay_init(&global.replay);
 	player_init(&global.plr);
+	stats_init(&global.plr.stats);
```

Comment:
I think you also need this call in `main_singlestg_begin_game` (`main.c`). That is where game initialization is done when starting a stage from the command line (`taisei --play --sid <stageID>`).

(Ideally we should really improve our terrible game-session abstraction (or lack thereof), but that's out of scope for this PR)